src/actions/ui.js

```diff
@@ -122,7 +123,11 @@ export const tClearUiProgramStageDimensions =
         if (needsClearing) {
             const idsToRemove = state.ui.layout.columns
                 .concat(state.ui.layout.filters)
-                .filter((id) => id.includes(`${stageId}.`))
+                .filter(
+                    (dimensionId) =>
+                        dimensionId.includes(`${stageId}.`) ||
+                        dimensionId === DIMENSION_ID_SCHEDULED_DATE
```

Comment:
This is fine for now, but we should ideally centralize all the "rules" somewhere to make them easy to keep track of and come back and update later.